WcaOnRails/spec/controllers/api_competitions_controller_spec.rb

```diff
@@ -40,4 +40,234 @@
       expect(parsed_body["website"]).to eq "http://example.com"
     end
   end
+
+  describe 'GET #index' do
+    it 'sorts newest to oldest' do
+      yesteryear_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.year.ago)
+      yesterday_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.ago)
+      today_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 0.days.ago)
+      tomorrow_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.from_now)
+
+      get :index
+      expect(response.status).to eq 200
+      json = JSON.parse(response.body)
+      expect(json.map { |c| c["id"] }).to eq [tomorrow_comp, today_comp, yesterday_comp, yesteryear_comp].map(&:id)
```

Comment:
Cool! Just to be clear: the `v0` refers to the WCA website API, whereas `1.0 alpha` refers to the WCIF data format. Either way, both of them are in flux. The only people who know about these things are "us" (the WCA Software Team) and anyone who asks us about them.